src/main/java/de/tum/in/www1/artemis/service/connectors/ldap/LdapAuthenticationProvider.java

```diff
@@ -58,41 +61,101 @@ public Authentication authenticate(Authentication authentication) throws Authent
+        }
+        else {
+            // It's a login, try to find the LDAP user in the external user management system based on the given login
+            ldapUserDto = ldapUserService.findByLogin(loginOrEmail).orElseThrow(() -> new BadCredentialsException("Wrong credentials"));
+        }
 
-        final LdapUserDto ldapUserDto = ldapUserService.findByUsername(username).orElseThrow(() -> new BadCredentialsException("Wrong credentials"));
-
-        log.debug("Finished ldapUserService.findByUsername in {}", TimeLogUtil.formatDurationFrom(start));
+        if (isEmail && optionalUser.isEmpty()) {
+            // this is an edge case which could happen when the user email changed or the user has multiple email addresses and used a secondary email to login
+            // therefore, double check if the Artemis User with the LDAP login (based on the given email) exists. If yes, we will use this user and update the LDAP values below
+            // without this code a second user would be created in Artemis which is not what we want (additionally this would fail because of unique constraints)
+            optionalUser = userRepository.findOneWithGroupsAndAuthoritiesByLogin(ldapUserDto.getLogin());
+        }
+        log.debug("Finished ldapUserService.findByLogin in {}", TimeLogUtil.formatDurationFrom(start));
         start = System.nanoTime();
 
-        // We create our own authorization and use the credentials of the user.
+        // Use the given password to compare it with the LDAP entry (i.e. check the authorization)
         byte[] passwordBytes = Utf8.encode(password);
         boolean passwordCorrect = ldapTemplate.compare(ldapUserDto.getUid().toString(), "userPassword", passwordBytes);
-        log.debug("Compare password with LDAP entry for user {} to validate login", username);
-        // this is the normal case, where the password is validated
+        log.debug("Compare password with LDAP entry for user {} to validate login", loginOrEmail);
         if (!passwordCorrect) {
             throw new BadCredentialsException("Wrong credentials");
         }
 
         log.debug("Finished ldapTemplate.compare password in {}", TimeLogUtil.formatDurationFrom(start));
 
-        return optionalUser.orElseGet(() -> {
-            User newUser = userCreationService.createUser(ldapUserDto.getUsername(), null, null, ldapUserDto.getFirstName(), ldapUserDto.getLastName(), ldapUserDto.getEmail(),
+        // update the user details from ldapUserDto (because they might have changed, e.g. when the user changes the name)
+        if (optionalUser.isPresent()) {
+            User user = optionalUser.get();
+            boolean saveNeeded = false;
+            if (Objects.equals(user.getLogin(), ldapUserDto.getLogin())) {
+                user.setLogin(ldapUserDto.getLogin());
+                saveNeeded = true;
+            }
```

Comment:
```suggestion
            if (!Objects.equals(user.getLogin(), ldapUserDto.getLogin())) {
                user.setLogin(ldapUserDto.getLogin());
                saveNeeded = true;
            }
```
I think this should be inverted since setting it to an equal value will not do anything

Reply:
thanks 👍